### PR TITLE
CRM-16796 - Add ACL support for profile related api's

### PR DESCRIPTION
**Pull request by jitendrapurohit**
https://issues.civicrm.org/jira/browse/CRM-16796

### CODE REVIEW

File: Civi/API/Subscriber/PermissionCheck.php [hunk 1]
```diff
@@ -87,4 +87,26 @@ public function onApiAuthorize(\Civi\API\Event\AuthorizeEvent $event) {
     }
   }
 
+  /**
+   * check API for ACL permission.
+   * @param $apiRequest
```
Comment (eileenmcnaughton) [inline Civi/API/Subscriber/PermissionCheck.php]: This is really trivial so you don't need to submit a new commit just for this- but if we were enforcing more of the PHPCS rules then the comment block would look like:

  /**
  *  Check API for ACL permission.  
  *
  * @param array $apiRequest
  * 
  * @ return bool
  */

ie. cap in first word & line breaks around the @param declarations. (also $apiRequest is an array)

Comment (jitendrapurohit) [inline Civi/API/Subscriber/PermissionCheck.php]: @eileenmcnaughton Thanks for your suggestion. I've updated the comment block as per the rules.

**Comment by eileenmcnaughton** [comment]
Tests look good

**Comment by davecivicrm** [comment]
Needs to be rebased, then ok to merge (conflict due to new comment block added on a test).

**Comment by totten** [comment]
jenkins, test this please

**Pull request merged**